FreeRTOS/Demo/shakti/bsp/core/init.c

```diff
@@ -0,0 +1,164 @@
+ */
+static void section_init(void)
+{
+#if 0
+	/*Enable below code only on need
+	 */
+	while(heap_start<=heap_end)
+	{
+		*heap_start=0xff;
+		heap_start++;
+	}
+
+	while(stack_end>=stack_start)
+	{
+		*stack_end=0x0;
+		stack_end--;
+	}
+#endif
+
+}
+
+/** @fn void trap_init()
+ * @brief Initialize the trap/interrupt callback routines with user defined handler.
+ * @details Assign default handler for trap / interrupt that does not have user defined
+ *          callback routines"
+ */
+static void trap_init(void)
+{
+	log_trace("trap_init entered \n ");
+
+	mcause_interrupt_table[USER_SW_INTERRUPT]        = default_handler;
+	mcause_interrupt_table[SUPER_SW_INTERRUPT]       = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT0]      = default_handler;
+	mcause_interrupt_table[MACH_SW_INTERRUPT]        = default_handler;
+	mcause_interrupt_table[USER_TIMER_INTERRUPT]     = default_handler;
+	mcause_interrupt_table[SUPER_TIMER_INTERRUPT]    = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT1]      = default_handler;
+	mcause_interrupt_table[MACH_TIMER_INTERRUPT]     = mach_clint_handler;
+	mcause_interrupt_table[USER_EXT_INTERRUPT]       = default_handler;
+	mcause_interrupt_table[SUPERVISOR_EXT_INTERRUPT] = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT2]      = default_handler;
+	mcause_interrupt_table[MACH_EXTERNAL_INTERRUPT]  = mach_plic_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT3]      = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT4]      = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT5]      = default_handler;
+	mcause_interrupt_table[RESERVED_INTERRUPT6]      = default_handler;
+
+	mcause_trap_table[INSTRUCTION_ADDRESS_MISALIGNED] =
+		default_handler;
+	mcause_trap_table[INSTRUCTION_ACCESS_FAULT] =
+		default_handler;
+	mcause_trap_table[ILLEGAL_INSTRUCTION] =
+		default_handler;
+	mcause_trap_table[BREAKPOINT] =
+		default_handler;
+	mcause_trap_table[LOAD_ADDRESS_MISALIGNED] =
+		default_handler;
+	mcause_trap_table[LOAD_ACCESS_FAULT] =
+		default_handler;
+	mcause_trap_table[STORE_AMO_ADDRESS_MISALIGNED] =
+		default_handler;
+	mcause_trap_table[STORE_AMO_ACCESS_FAULT] =
+		default_handler;
+	mcause_trap_table[ENVIRONMENT_CALL_FROM_U_MODE] =
+		default_handler;
+	mcause_trap_table[ENVIRONMENT_CALL_FROM_S_MODE] =
+		default_handler;
+	mcause_trap_table[RESERVED_TRAP1] =
+		default_handler;
+	mcause_trap_table[ENVIRONMENT_CALL_FROM_M_MODE] =
+		default_handler;
+	mcause_trap_table[INSTRUCTION_PAGE_FAULT] =
+		default_handler;
+	mcause_trap_table[LOAD_PAGE_FAULT] =
+		default_handler;
+	mcause_trap_table[RESERVED_TRAP2] =
+		default_handler;
+	mcause_trap_table[STORE_AMO_PAGE_FAULT] =
+		default_handler;
+
+	log_trace("trap_init exited \n ");
+}
+
+/** @fn void init(void)
+ * @brief initialize the necessary variables for system start
+ */
+void init(void)
+{
+//	section_init(); // uncomment on need basis
```

Comment:
The content of the function this called is hidden " #if 0"

Reply:
Yes this is intentional. If someone wants it they can enable it. It waste some clock cycles on setting memory to zero.